1664481: match syspurpose attributes case insensitively [ENT-1284]

nikosmoum opened the pull request:
- During auto-attach and syspurpose status calculations, syspurpose
  attribute values are now compared with case insensitivity.

Ceiu commented on server/src/main/java/org/candlepin/policy/SystemPurposeComplianceRules.java:
```diff
@@ -168,8 +161,11 @@ public SystemPurposeComplianceStatus getStatus(Consumer consumer,
-                    if (roles.contains(consumer.getRole())) {
-                        status.addCompliantRole(consumer.getRole(), entitlement);
+
+                    final String unsatisfedRoleFinalCopy = unsatisfedRole;
```
Do we still need the final ref with lambdas? I thought I remembered reading that they fixed that part in Java9+.

nikosmoum replied:
I am not aware of such a fix, but aren't we using java 8 still anyway?

Ceiu replied:
Right... bit of a derp moment here for me.

I just tried googling it and nothing immediately came up. Maybe wishful thinking.

Ceiu replied:
Scratch that. I just found that in Java8 they introduced something called "effectively final" -- or a variable that doesn't change after initialization. That might apply here, as the unsatisifiedRole doesn't change during the loop, and technically is redeclared on every iteration.

nikosmoum replied:
Unfortunately it doesn't apply here. `unsatisifiedRole` is not effectively final because if we get a match for it, it is re-assigned to null (and this is required because after every iteration, we check for it: `StringUtils.isNotEmpty(unsatisfedRole)`. We still need to create/assign `unsatisfedRoleFinalCopy`, but we don't need the `final` identifier.

Ceiu replied:
Yeah. If we have to redeclare it for the sake of a lambda, I'm okay with leaving it final, as that's a pretty well-known pattern for such things.

nikosmoum commented:
@Ceiu I've moved the trim calls to the product attribute list building, and also added trim calls to the consumer's side of attribute values. I added unit tests for these usecases in `SystemPurposeComplianceRulesTest`.
